client/src/dataProviders/publiclyDataProviders.js

```diff
@@ -1,12 +1,16 @@
 // @flow
-import {setEntities} from '../actions/verejneActions'
+import React from 'react'
+import {setEntities, setEntitySearchEids} from '../actions/verejneActions'
 
 import type {Entity} from '../state'
 import type {Dispatch} from '../types/reduxTypes'
 
 const dispatchEntities = () => (ref: string, data: Array<Entity>, dispatch: Dispatch) =>
   dispatch(setEntities(data))
 
+const dispatchSearchEids = () => (ref: string, data: Array<{eid: string}>, dispatch: Dispatch) =>
```

Comment:
I'm strongly in favor of having raw data in state in the form of 'object maps' ({id1: entity1, id2: entity2}) - and then when you need to work with an array, just have a selector which gets you Object.values of said map

Reply:
I know this format would be great but are you sure it can be easily achieved here? (with data providers)? 

Problem is the api only returns the `eids`, and you need additional request to get info for each `eid`. The problem is that you cannot modify the same part of the state on which rely different data providers (because both will try to refetch their data..). For this reason I fetch only the `eids` into state, and then use a component which will download and show the info for the eid... 

Although we could store the `eids` in local state and then download the info for each eid and put them into redux... But I don't see any added value here.

Also we talked about this with Rasto, that the api should return the info directly (not just the eid), so I think we might change it after the api change

Reply:
ok, makes sense, I did not realize when writing the previous comment that this part concerns just the ids and not ids together with objects (sorry!)